go/vt/vtgate/endtoend/sequence_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endtoend
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"vitess.io/vitess/go/mysql"
+)
+
+func TestSeq(t *testing.T) {
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	//Why below query doesn't work for seq table?
+	//exec(t, conn, "insert into sequence_test_seq(id, next_id, cache) values(0,1,10)")
```

Comment:
@deepthi  Do you know why the above line will not work? 
I had to pass this insert as a setup DDL to make it work.

Reply:
What error did you get?

Reply:
I get below error: Is there any restriction that the seq table should be prepopulated before bringing the cluster? 

```
E1007 11:53:54.143075   17761 server.go:270] mysql_server caught panic:
runtime error: index out of range [0] with length 0
/usr/local/go/src/runtime/panic.go:75 (0x102f322)
	goPanicIndex: panic(boundsError{x: int64(x), signed: true, y: y, code: boundsIndex})
/Users/ajeetjain/ps/vitess/go/vt/vtgate/engine/insert.go:394 (0x1a27ca0)
	io/vitess/go/vt/vtgate/engine.(*Insert).getInsertShardedRoute: keyspaceIDs, err := ins.processPrimary(vcursor, vindexRowsValues[0], ins.Table.ColumnVindexes[0], bindVars)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/engine/insert.go:272 (0x1a25488)
	io/vitess/go/vt/vtgate/engine.(*Insert).execInsertSharded: rss, queries, err := ins.getInsertShardedRoute(vcursor, bindVars)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/engine/insert.go:222 (0x1a249f2)
	io/vitess/go/vt/vtgate/engine.(*Insert).Execute: return ins.execInsertSharded(vcursor, bindVars)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/executor.go:304 (0x1a780c8)
	io/vitess/go/vt/vtgate.(*Executor).handleExec: qr, err := plan.Instructions.Execute(vcursor, bindVars, true)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/executor.go:210 (0x1a769f7)
	io/vitess/go/vt/vtgate.(*Executor).execute: qr, err := e.handleExec(ctx, safeSession, sql, bindVars, destKeyspace, destTabletType, dest, logStats)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/executor.go:134 (0x1a7659b)
	io/vitess/go/vt/vtgate.(*Executor).Execute: result, err = e.execute(ctx, safeSession, sql, bindVars, logStats)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/vtgate.go:289 (0x1a9a741)
	io/vitess/go/vt/vtgate.(*VTGate).Execute: qr, err = vtg.executor.Execute(ctx, "Execute", NewSafeSession(session), sql, bindVariables)
/Users/ajeetjain/ps/vitess/go/vt/vtgate/plugin_mysql_server.go:206 (0x1a883d5)
	io/vitess/go/vt/vtgate.(*vtgateHandler).ComQuery: session, result, err := vh.vtg.Execute(ctx, session, query, make(map[string]*querypb.BindVariable))
/Users/ajeetjain/ps/vitess/go/mysql/conn.go:1070 (0x148e647)
	io/vitess/go/mysql.(*Conn).execQuery: err := handler.ComQuery(c, query, func(qr *sqltypes.Result) error {
/Users/ajeetjain/ps/vitess/go/mysql/conn.go:768 (0x148dec0)
	io/vitess/go/mysql.(*Conn).handleNextCommand: if err := c.execQuery(sql, handler, more); err != nil {
/Users/ajeetjain/ps/vitess/go/mysql/server.go:452 (0x14a8e0e)
	io/vitess/go/mysql.(*Listener).handle: err := c.handleNextCommand(l.handler)
/usr/local/go/src/runtime/asm_amd64.s:1357 (0x1060b30)
	goexit: BYTE	$0x90	// NOP
--- FAIL: TestSeq (0.01s)
    sequence_test.go:36: EOF (errno 2013) (sqlstate HY000) during query: insert into sequence_test_seq(id, next_id, cache) values(0,1,10)
I1007 11:53:54.149751   17761 engine.go:278]
```

Reply:
This is because sequence_test_seq is in a sharded keyspace, but no vindex is defined for it.